clcache.py

```diff
@@ -232,6 +232,9 @@ def getManifestHash(compilerBinary, commandLine, sourceFile):
         # One of the few exceptions to this rule is the /MP switch, which only
         # defines how many compiler processes are running simultaneusly.
         commandLine = [arg for arg in commandLine if not arg.startswith("/MP")]
+        baseDir = os.environ.get('CLCACHE_BASEDIR')
+        if baseDir:
+            commandLine = [arg.replace(baseDir, BASEDIR_REPLACEMENT) for arg in commandLine]
```

Comment:
You could reuse the existing `collapseBasedirToPlaceholder` here and use

```python
commandLine = [collapseBasedirToPlaceholder(arg) for arg in commandLine if not arg.startswith("/MP")]
```

in line 234.